### PR TITLE
fix icon for nav items with on_top: true

### LINKED ISSUE #4023: Impossible to set an icon for "on_top" element menu
Before making a PR I wanted to make sure I'm not missing something:
when setting the option 'on_top' to true on an element of the menu, it is impossible to set an icon for this element:
#### Sonata packages

``` bash
$ composer show sonata-project/*
sonata-project/admin-bundle              3.4.0                    The missing Symfony Admin Generator
sonata-project/block-bundle              3.1.1                    Symfony SonataBlockBundle
sonata-project/cache                     1.0.7                    Cache library
sonata-project/core-bundle               3.0.3                    Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.2                      Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2                    Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.0.5                    Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.1.10                   Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.5.0                    Lightweight Exporter library
sonata-project/google-authenticator      1.0.2                    Library to integrate Google Authenticator into a PHP project
sonata-project/user-bundle               dev-rc_sf3tweaks bf121be Symfony SonataUserBundle
```
## Steps to reproduce

Fresh install of SonataAdmin, add the following config:

```yaml
sonata_admin:
  dashboard:
      groups:
          home:
            on_top: true
            icon: '<i class="fa fa-home"></i>'
            items:
              - route: sonata_admin_dashboard
                label: home
```
## Expected results

A nice "home" icon next to the element menu "home" in the left menu.
## Actual results

"Home" is there, but the icon is not (it is not a FontAwesome problem, the html is empty).
## Explanation

In sonata-project/admin-bundle/Resources/views/Menu/sonata_menu.html.twig:

``` twig
{% set icon = item.attribute('icon')|default(item.level > 1 ? '<i class="fa fa-angle-double-right"></i>' : '') %}
```

The html code for the icon is in this icon variable, but item.attribute() doesn't work.

Either my config is wrong or this line should be:

``` twig
{% set icon = item.extra('icon')|default(item.level > 1 ? '<i class="fa fa-angle-double-right"></i>' : '') %}
```

### PR DESCRIPTION
**Pull request by OskarStark**
I am targeting this branch, because this is BC

Refs #4023

## Changelog

```markdown
### Fixed
- Show icon for nav items when using `on_top`-option 
```

## Subject

As mentioned here https://github.com/sonata-project/SonataAdminBundle/issues/4023#issuecomment-302629065, I can confirm this fixes the problem

### CODE REVIEW

**Comment by greg0ire** [comment]
Please sum up #4023 in the body of the commit message

**Comment by OskarStark** [comment]
> Please sum up #4023 in the body of the commit message

can you elaborate?

**Comment by greg0ire** [comment]
Just explain what is wrong and how your commit fixes that :)

**Comment by OskarStark** [comment]
done

**Comment by greg0ire** [comment]
Please fix the "pe puplated" typo

**Comment by OskarStark** [comment]
> Please fix the "pe puplated" typo

done

**Comment by greg0ire** [comment]
Is "puplated" a real word?

**Comment by OskarStark** [comment]
> Is "puplated" a real word?

ofc not...damn...fixed 😄

File: Resources/views/Menu/sonata_menu.html.twig [hunk 1]
```diff
@@ -27,7 +27,7 @@
     {% spaceless %}
         {% if item.extra('on_top') is defined and not item.extra('on_top') %}
             {% set translation_domain = item.extra('translation_domain', 'messages') %}
-            {% set icon = item.attribute('icon')|default(item.level > 1 ? '<i class="fa fa-angle-double-right" aria-hidden="true"></i>' : '') %}
+            {% set icon = item.extra('icon')|default(item.level > 1 ? '<i class="fa fa-angle-double-right"></i>' : '') %}
```
Comment (core23) [inline Resources/views/Menu/sonata_menu.html.twig]: Why did you remove ` aria-hidden="true"`?

Comment (OskarStark) [inline Resources/views/Menu/sonata_menu.html.twig]: indeed...

restored

**Comment by greg0ire** [comment]
I restarted the build

**Comment by OskarStark** [comment]
strange...
screenshot 2017-05-23 12 22 19

**Comment by greg0ire** [comment]
Yeah I saw that. Maybe retry later, then look on the Composer issue tracker.

**Comment by greg0ire** [comment]
https://github.com/composer/composer/issues/6440

**Comment by OskarStark** [comment]
as this is a composer error, we can merge this IMO

**Comment by greg0ire** [comment]
@sonata-project/contributors please review

**Comment by jordisala1991** [comment]
it is consistent with the `else` part. But that `if` is used when not `on_top`, or am I missing something?

**Comment by OskarStark** [comment]
> it is consistent with the else part. But that if is used when not on_top, or am I missing something?

Yes I tested both

**Comment by jordisala1991** [comment]
Thanks @OskarStark
